### PR TITLE
Upgrade React and Enzyme

### PR DESCRIPTION
**Pull request by jackrehaag**
### Changes
`react` `15.3.2` -> `16.0.0`
`enzyme` `2.4.1` -> `3.3.0`

Both upgrades include some breaking changes, code changes have been made to account for these.

## Code changes related to React upgrade
- Using `PropTypes` rather than `React.PropTypes` and inclusion of `prop-types` package
https://reactjs.org/blog/2017/04/07/react-v15.5.0.html#migrating-from-reactproptypes
- Removed `react-addons-test-utils` due to deprecating, did not replace with `react-dom/test-utils` as `TestUtils` is not actually used in the codebase

## Code changes related to Enzyme upgrade
- Inclusion of `enzyme-adapter-react-16` package
https://github.com/airbnb/enzyme#upgrading-from-enzyme-2x-or-react--16
- Use Enzyme's `mount` instead of `shallow` https://github.com/airbnb/enzyme/blob/master/docs/guides/migration-from-2-to-3.md

#### Trello Ticket
https://trello.com/c/eVYd6AX4/173-update-key-dependencies

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -77,19 +78,17 @@
...
     "pre-git": "^3.10.0",
     "prettier": "^1.8.2",
+    "prop-types": "^15.5.7",
     "proxyquire": "^1.8.0",
     "ramda": "^0.22.1",
     "raw-loader": "^0.5.1",
-    "react": "^15.3.2",
-    "react-addons-test-utils": "^15.3.2",
-    "react-dom": "^15.3.1",
+    "react": "^16.0.0",
```
Comment (dpiatek) [inline package.json]: Looks like latest stable is 16.3.1 - can we not update to that?

File: site/components/avatar/test.js [hunk 2]
```diff
@@ -1,9 +1,12 @@
 // @flow
 import React from 'react';
 import { expect } from 'chai';
-import { shallow } from 'enzyme';
+import Enzyme, { shallow } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
 import Avatar from '.';
 
+Enzyme.configure({ adapter: new Adapter() });
```
Comment (dpiatek) [inline site/components/avatar/test.js]: Can this not go into `mocha-env.js` and just be called there once?

**Comment by dpiatek** [comment]
Looking good! Think it needs a rebase ... did not realize we have so many propTypes left (flow should really be used there).

**Comment by jackrehaag** [comment]
@dpiatek ♻️

File: site/components/component-renderer/title3.js [hunk 3]
```diff
@@ -11,6 +11,6 @@ export default function Title3({ children }: Title3Props) {
 }
 /*
 Title3.propTypes = {
-  children: React.PropTypes.node,
+  children: PropTypes.node,
 };
 */
```
Comment (dpiatek) [inline site/components/component-renderer/title3.js]: why do we need this commented out code?

File: site/components/content/index.js [hunk 4]
```diff
@@ -10,6 +10,6 @@ export default function Content({ children }: ContentProps) {
 }
 /*
 Content.propTypes = {
-  children: React.PropTypes.node,
+  children: PropTypes.node,
 };
```
Comment (dpiatek) [inline site/components/content/index.js]: As above

File: site/components/grid/index.js [hunk 5]
```diff
@@ -35,10 +36,10 @@ type GridProps = {
...
   static propTypes = {
-    breakOn: React.PropTypes.string,
-    children: React.PropTypes.node,
-    extraClassName: React.PropTypes.string,
-    fit: React.PropTypes.bool,
+    breakOn: PropTypes.string,
+    children: PropTypes.node,
+    extraClassName: PropTypes.string,
+    fit: PropTypes.bool,
   };
```
Comment (dpiatek) [inline site/components/grid/index.js]: this propType is not needed, the exact same thing is covered by the flow type above

File: site/components/wall/index.js [hunk 6]
```diff
@@ -1,8 +1,9 @@
 // @flow
 import * as React from 'react';
+import PropTypes from 'prop-types';
 import { Grid, Cell } from '../grid';
 
-const { node, number } = React.PropTypes;
+const { node, number } = PropTypes;
```
Comment (dpiatek) [inline site/components/wall/index.js]: propTypes can be removed from this file

File: site/pages/about-us/principles-slice/principle/index.js [hunk 7]
```diff
@@ -14,9 +15,9 @@ const Principle = ({ number, title, description }) => (
 );
 
 Principle.propTypes = {
-  number: React.PropTypes.string,
-  title: React.PropTypes.string,
-  description: React.PropTypes.string,
+  number: PropTypes.string,
+  title: PropTypes.string,
+  description: PropTypes.string,
```
Comment (dpiatek) [inline site/pages/about-us/principles-slice/principle/index.js]: This propType should be migrated to a flow type (and an `// @flow` declaration should be added on top of the file

File: site/pages/about-us/timeline-slice/content/index.js [hunk 8]
```diff
@@ -39,13 +40,13 @@ const Content = ({ year, title, text, fact, image, flip }) => {
...
-  image: React.PropTypes.string.isRequired,
-  mobileImage: React.PropTypes.string.isRequired,
-  flip: React.PropTypes.bool,
+  year: PropTypes.string.isRequired,
+  title: PropTypes.string.isRequired,
+  text: PropTypes.string.isRequired,
+  fact: PropTypes.string.isRequired,
+  image: PropTypes.string.isRequired,
+  mobileImage: PropTypes.string.isRequired,
+  flip: PropTypes.bool,
```
Comment (dpiatek) [inline site/pages/about-us/timeline-slice/content/index.js]: + to be migrated

File: site/pages/about-us/timeline-slice/mobileImage/index.js [hunk 9]
```diff
@@ -40,8 +41,8 @@ const MobileImage = ({ index, onChangeIndex }) => {
 };
 
 MobileImage.propTypes = {
-  index: React.PropTypes.number.isRequired,
-  onChangeIndex: React.PropTypes.func.isRequired,
+  index: PropTypes.number.isRequired,
+  onChangeIndex: PropTypes.func.isRequired,
```
Comment (dpiatek) [inline site/pages/about-us/timeline-slice/mobileImage/index.js]: + to be migrated

File: site/pages/about-us/timeline-slice/navigator/arrow/index.js [hunk 10]
```diff
@@ -31,9 +32,9 @@ const Arrow = ({ direction, onClick, currentIndex }) => {
 };
 
 Arrow.propTypes = {
-  direction: React.PropTypes.string.isRequired,
-  onClick: React.PropTypes.func.isRequired,
-  currentIndex: React.PropTypes.number.isRequired,
+  direction: PropTypes.string.isRequired,
+  onClick: PropTypes.func.isRequired,
+  currentIndex: PropTypes.number.isRequired,
```
Comment (dpiatek) [inline site/pages/about-us/timeline-slice/navigator/arrow/index.js]: + to be migrated

File: site/pages/about-us/timeline-slice/navigator/index.js [hunk 11]
```diff
@@ -24,8 +25,8 @@ const TimelineNav = ({ currentIndex, onClick }) => {
 };
 
 TimelineNav.propTypes = {
-  currentIndex: React.PropTypes.number.isRequired,
-  onClick: React.PropTypes.func.isRequired,
+  currentIndex: PropTypes.number.isRequired,
+  onClick: PropTypes.func.isRequired,
```
Comment (dpiatek) [inline site/pages/about-us/timeline-slice/navigator/index.js]: + to be migrated

File: site/pages/about-us/timeline-slice/navigator/item/index.js [hunk 12]
```diff
@@ -21,9 +22,9 @@ const TimelineNav = ({ value, onClick, currentIndex }) => {
 };
 
 TimelineNav.propTypes = {
-  value: React.PropTypes.number.isRequired,
-  currentIndex: React.PropTypes.number.isRequired,
-  onClick: React.PropTypes.func.isRequired,
+  value: PropTypes.number.isRequired,
+  currentIndex: PropTypes.number.isRequired,
+  onClick: PropTypes.func.isRequired,
```
Comment (dpiatek) [inline site/pages/about-us/timeline-slice/navigator/item/index.js]: + to be migrated

File: site/pages/events/events-timeline-title/index.js [hunk 13]
```diff
@@ -1,5 +1,6 @@
 // @flow
-import React, { PropTypes } from 'react';
+import React from 'react';
+import PropTypes from 'prop-types';
```
Comment (dpiatek) [inline site/pages/events/events-timeline-title/index.js]: + to be migrated

File: site/pages/job/index.js [hunk 14]
```diff
@@ -61,9 +62,9 @@ export default function Job({ job }: JobProps) {
...
 
 Job.propTypes = {
-  job: React.PropTypes.shape({
-    title: React.PropTypes.string,
-    fullDescription: React.PropTypes.string,
-    applicationUrl: React.PropTypes.string,
+  job: PropTypes.shape({
+    title: PropTypes.string,
+    fullDescription: PropTypes.string,
+    applicationUrl: PropTypes.string,
```
Comment (dpiatek) [inline site/pages/job/index.js]: this can be removed

File: site/pages/technology/newsletter-slice/after-sign-up/index.js [hunk 15]
```diff
@@ -1,9 +1,10 @@
 import React, { Component } from 'react';
+import PropTypes from 'prop-types';
 import classnames from 'classnames/bind';
 import styles from '../style.css';
 
 const cx = classnames.bind(styles);
-const { func, string, bool } = React.PropTypes;
+const { func, string, bool } = PropTypes;
```
Comment (dpiatek) [inline site/pages/technology/newsletter-slice/after-sign-up/index.js]: + to be migrated

File: site/pages/technology/newsletter-slice/before-sign-up/index.js [hunk 16]
```diff
@@ -100,7 +101,7 @@ class BeforeSignUp extends Component {
   }
 }
 
-const { func, string } = React.PropTypes;
+const { func, string } = PropTypes;
```
Comment (dpiatek) [inline site/pages/technology/newsletter-slice/before-sign-up/index.js]: + to be migrated

File: site/slices/tech-slice/index.js [hunk 17]
```diff
@@ -1,4 +1,5 @@
-import React, { PropTypes } from 'react';
+import React from 'react';
+import PropTypes from 'prop-types';
```
Comment (dpiatek) [inline site/slices/tech-slice/index.js]: + to be migrated

**Comment by jackrehaag** [comment]
@dpiatek One more ♻️ please

**Pull request merged**